gnucash/import-export/import-main-matcher.c

```diff
@@ -925,17 +925,49 @@ match_func (GtkEntryCompletion *completion, const char *entry_str,
     return ret;
 }
 
+typedef struct
+{
+    GtkWidget *parent;
+    GtkWidget *entry;
+    GtkWidget *override_edit;
+    bool *sensitive;
+} OverrideData;
+
+static void override_edit_clicked (GtkWidget *widget, OverrideData *data)
+{
+    if (!gnc_verify_dialog (GTK_WINDOW (data->parent), false, "WARNING: %s",
+                            "This will rename non-identical data. Continue?"))
+        return;
+    gtk_widget_set_sensitive (data->override_edit, false);
+    gtk_widget_set_visible (data->override_edit, false);
+    gtk_widget_set_sensitive (data->entry, true);
+    gtk_entry_set_text (GTK_ENTRY (data->entry), "");
+    gtk_widget_grab_focus (data->entry);
+    *(data->sensitive) = true;
+}
+
 static void
-setup_entry (GtkWidget *entry, bool sensitive, GHashTable *hash,
-             const char *initial)
+setup_entry (GtkWidget *entry, GtkWidget *override_edit, GNCImportMainMatcher *info,
+             bool* sensitive_ptr, GHashTable *hash, const char *initial)
 {
+    bool sensitive = *sensitive_ptr;
     gtk_widget_set_sensitive (entry, sensitive);
+    gtk_widget_set_visible (override_edit, !sensitive);
+    gtk_widget_set_sensitive (override_edit, !sensitive);
+
     if (sensitive && initial && *initial)
         gtk_entry_set_text (GTK_ENTRY (entry), initial);
     else if (!sensitive)
     {
+        OverrideData *override_data = g_new (OverrideData, 1);
+        override_data->parent = info->main_widget;
+        override_data->override_edit = override_edit;
+        override_data->entry = entry;
+        override_data->sensitive = sensitive_ptr;
         gtk_entry_set_text (GTK_ENTRY (entry), _("Disabled"));
-        return;
+        g_signal_connect (override_edit, "clicked", G_CALLBACK (override_edit_clicked),
+                          override_data);
+        info->new_strings = g_list_prepend (info->new_strings, override_data);
```

Comment:
`info->new_strings` is being reused to free OverrideData structs when import matcher completes. It could be renamed to `info->data_to_gfree`.